d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -261,13 +326,20 @@ static ResourceType fromTypeUrl(String typeUrl)
    * will always notify the given watcher of the current data if it is already present, even if the given watcher was
    * already subscribed to said resource. However, the subscription will only be added once.
    */
-  abstract void watchXdsResource(String resourceName, ResourceWatcher watcher);
+  public abstract void watchXdsResource(String resourceName, ResourceWatcher watcher);
+
+  /**
+   * Subscribes the given {@link WildcardResourceWatcher} to all the resources of the corresponding type. The watcher
+   * will be notified whenever a resource is added or removed. Repeated calls to this function with the same watcher
+   * will always notify the given watcher of the current data.
+   */
+  public abstract void watchAllXdsResources(WildcardResourceWatcher watcher);
 
-  abstract void startRpcStream();
+  public abstract void startRpcStream();
```

Comment:
Why do we need to make these methods public now? Looks like they weren't previously

Reply:
Because `watchXdsResource` is a public class, so all the abstract method should be `public`

Reply:
I see, so the XdsClient previously was wrong and we're fixing it here?

Reply:
Yes exactly